### ISSUE TITLE
[7.5][startailcoon] Settle orders issues

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-ui#744]
On this page: https://wallet.bitshares.org/#/market/BTS_CNY, data in the "settle orders" area has some issues:
* `price` column: currently it's showing settlement price (same to the one on top of the page), it's better to show `settlement_price * (1+settle_offset%)`, for example, current offset is 5% for bitCNY, settlement price is `0.691728 bitCNY/BTS`, so the price of settle orders should be `0.691728*105%=0.726314`.
* `quantity` column: showing only `quantity(BTS)` doesn't make sense, at least should show `bitCNY`, best if show both `bitCNY` and `BTS`.
* `settlement date` column: should deal with timezone. Currently it shows wrong date when timezone of the computer is not GMT. (By the way, I always don't like the descriptive format of time like `X minutes in the future or X hours ago, but prefer ISO time format which looks more accurate, best accurate to seconds).

**Comment by startailcoon** [comment]
Additionally we could color the settlement rows or header in the buy/sale color depending inbwhatvside of the book you're looking at. Since we've done so with the depth graph in another issue. 

If we add both assets we could hide something when the table is to small. Shouldn't be an issue.

**Comment by wmbutler** [comment]
I had a suggestion to allow My Trades, Market Trades and Sell Orders span double width as tabs in the same container. If we also incorporated Settle Orders, this would give us room for the additional column you are proposing as well.

**Comment by wmbutler** [comment]
This needs to be claimed.

**Comment by startailcoon** [comment]
I can claim this for the next milestone

**Comment by wmbutler** [comment]
@startailcoon

**Comment by wmbutler** [comment]
@startailcoon are you planning to have this complete?

**Comment by startailcoon** [comment]
> I had a suggestion to allow My Trades, Market Trades and Sell Orders span double width as tabs in the same container. If we also incorporated Settle Orders, this would give us room for the additional column you are proposing as well.

I assume here that you're referring to the `My Open Orders` on the exchange tab, and not the actual markets sell orders.

It would mean we have something like this (crudely demonstrated with tables here)

&nbsp;|&nbsp;
-|-
Buy (market)|Sell (market)
Buy Orders|Sell Orders
My Open Orders/My Trades/Market Trades/Settle Orders|(<- with tabs spanning twice)

I would make `My Open Orders` the default tab to view.

**Comment by wmbutler** [comment]
> I assume here that you're referring to the My Open Orders on the exchange tab, and not the actual markets sell orders.

Correct, I think the market sell orders should sit separately and I'd imagine most who have the opportunity move this to the left side of the screen.

**Comment by startailcoon** [comment]
# Question
As we're condensing 3 tables in to one larger, I've got two suggestions on how to manage the layout.

## Layout 1
Stack Buy/Sell orders over each other in xlarge wide mode. 
The new table will go in the empty spot.
When we go smaller the layout will go to the same as Layout 2 below.

In this layout we could switch place of `Price` and `Total` so it's more fluent. 

![marketwide_1](https://user-images.githubusercontent.com/12114550/34743887-a613cc14-f58b-11e7-95ce-bf8c0bbdac57.PNG)

## Layout 2
Widen everything so it fills out the whole screen even when we use xlarge wide mode.
The table will just schrink with the size of the window.
The new table will go below this.

![marketwide_2](https://user-images.githubusercontent.com/12114550/34743915-be4e7720-f58b-11e7-8ee3-c7a974ab4506.PNG)

## Horizontal Buy/Sell in Both Layouts
The horizontal layout for Buy/Sell will stay the same and look something like this, where the empty area is the space where the new table will be.

![marketwide_2_vert](https://user-images.githubusercontent.com/12114550/34743901-b0173610-f58b-11e7-9460-103a2c4965c5.PNG)

**Comment by svk31** [comment]
I'm not sure I like the idea of collapsing even more tables into one big one, since it leaves us with less information available at a glance than we had before.

The current layout is the result of lots and lots of iterations back and forth with feedback from users.

**Comment by startailcoon** [comment]
I tend to agree with svk, it could be a bit to much clicking to see the data that we now have available at a glance. 

The issue with more data in the settle table still stands. How can we supply more info in there.

Should we widen the table to allow more data, and hide some when it's to small?

**Comment by wmbutler** [comment]
It's important to determine what we actually need at a glance. As long as the last tab viewed is active, seems that this could work out fine for most users. I think a bigger problem is the duplication of columns and the lack of horizontal space to display everything in complete form.

**Comment by startailcoon** [comment]
Just to track this issue a little better, here is what I have at present.

### Clear Changes
- [ ] Include settlement offset percentage in the price
- [ ] Display both assets quantity
  - This enables to remove the asset label, saving some space. But it needs to have room for two assets with up to 8 decimals.
- [ ] Display short settlement date (DD/MM HH:MM:SS) and make sure it's timezone adjusted.
  - Will likely take more space than current `in x hours`

### To Solve
- Width of table to cope with displaying both assets full value even in small screens
  1. Collapsing several tables to one large?
  2. Widen `Settle Orders` to fit more data?

### Propsed changes
- Collapsing `My Open Orders`, `My Trade History`, `Market History`, and `Settle Orders` to one tab.

**Comment by wmbutler** [comment]
> Should we widen the table to allow more data, and hide some when it's to small?

Would prefer you don't go this route.

**Comment by wmbutler** [comment]
> Collapsing My Open Orders, My Trade History, Market History, and Settle Orders to one tab.

Having a hard time visualizing this.

**Comment by startailcoon** [comment]
> Would prefer you don't go this route.

We need some way of solving this, that I can work with.

> Having a hard time visualizing this.

I can take time to put this together, if it is what we want.


> As long as the last tab viewed is active, seems that this could work out fine for most users.

Which is what I intend to do, yes. The active tab the user has open, will be the default when a user opens a market again. No previous setting will need a default tab, I guess `My History` is as good as any.

**Comment by wmbutler** [comment]
Possibly add a count (3) (2) in tabs of:

- My Open Orders
- Settle Orders

This will provide insight into whether the user even needs to look at the tab. This could help mitigate the concern for having every tab visible. A long-term vision for this might include breakout tabs.

**Comment by startailcoon** [comment]
What does the number represents? Total items in the tab?

**Comment by startailcoon** [comment]
A few previews without data in them

# X-Large View
![image](https://user-images.githubusercontent.com/12114550/34905462-52aa6c4c-f859-11e7-98ef-6dc20e2980b4.png)

# X-Large View with Horizontal Trading 
![image](https://user-images.githubusercontent.com/12114550/34905468-6e199534-f859-11e7-950b-d984d47f5b4a.png)

# Narrow View
![image](https://user-images.githubusercontent.com/12114550/34905478-ad0b7d16-f859-11e7-91f1-4dbdda2ec27d.png)


# Narrow View with Horizontal Trading
![image](https://user-images.githubusercontent.com/12114550/34905473-9e5b6a7e-f859-11e7-8c07-58d629db9561.png)

**Comment by wmbutler** [comment]
Looking pretty good.

**Comment by startailcoon** [comment]
![bitshares-settlement](https://user-images.githubusercontent.com/12114550/34916352-64db0956-f937-11e7-9f74-2585d59416d2.gif)

**Comment by svk31** [comment]
I'm sorry but this is a step backwards for usability. While the original layout allowed me to see everything at a glance, I now need to both scroll down and click through different tabs to get the same information. Having three columns on wide screens is important imo.

Before:
![image](https://user-images.githubusercontent.com/6890015/34936355-5ef457d6-f9e1-11e7-8034-dfb1580ddea8.png)

After:
![image](https://user-images.githubusercontent.com/6890015/34936310-2f8565f8-f9e1-11e7-9c43-23c2084e3bce.png)

**Comment by startailcoon** [comment]
@billbutler please advise what the heading of this issue is and how to go further. I've spent a few hours on this task as requested.

**Comment by wmbutler** [comment]
@svk31 as a compromise here, do you think it makes sense to adjust the foundation settings to the new tabbed pane will appear to the right of the buy/sell order book?

- Note: You can't see settle order with the new layout or the existing layout because it ends up on the third row.
- We added a number indicator to make it clear whether there was relevant information inside My Open Orders
- Market Orders / My Orders already had a tabbed interface so we didn't modify that.

My only other thought here is that we had an interesting PR that proposed placing settle orders on the right side under the market selections. We could accept that PR and then let this be a three tabbed pane instead.

**Comment by svk31** [comment]
My primary concern is that having only two columns on large screens like mine just looks silly and is a waste of space.

**Comment by startailcoon** [comment]
> My primary concern is that having only two columns on large screens like mine just looks silly and is a waste of space.

I agree with that, thus why I suggested my proposal on `Layout 1`, where we put Buy/Sell in horizontal mode, (or we can put the Buy/Sell Forms, either would suffice) and have the tabbed layout on row 2. Since the tabbed layout requires at least 2 boxes in space to accommodate the extra fields for settlement. I still think that this would work better, for you as well @svk31.

I wide mode, we would then have:

```
Buy Form | Buy | Sell
Sell Form | Tabbed Section
```

or

```
Buy | Buy Form | Sell Form
Sell | Tabbed Section
```

any combination with the tabbed section using two sections in width is a good practice.
The first one would be a good choice, since it would put the market orders side by side on horizontal mode in all layouts down to one box per row in small screens.

**Comment by svk31** [comment]
The orderbook should be horizontal for easy comparison the way it is now, I don't like the idea of stacking them on top of each other. No one else does that and I don't think we should either.

**Comment by startailcoon** [comment]
> @startailcoon I do not think that Layout 1 will satisfy

>  I don't like the idea of stacking them on top of each other

Layout 1 can have the Buy/Sale stacked instead of the order book. 

But I assume your aware of that.

What if we just add Settle Orders together with today's My Open Orders, and hope that the extra fields of settle works well? 

This issue is more about settle orders than anything, and I'd like to have the changes in there. 

That way we can get rid of the settle orders being on the third row as well.

**Comment by wmbutler** [comment]
I think settle orders below the market selection is a reasonable compromise.

**Comment by startailcoon** [comment]
Looking on the proposed PR that was made #987, the space in the right hand menu is very narrow and wouldn't allow us to specify both trading pairs, and the longer date.

I still think adding it to another tab of the current boxes, like `My Orders`, or `My Trades/Market Trades`, is more reasonable than adding it to the right menu.

On my screen today, a laptop, it would be more convenient to move the Settle Orders in to My Open Orders. The right menu would be to small.

![image](https://user-images.githubusercontent.com/12114550/34961533-e96fcb30-fa3f-11e7-80c1-e219d768de00.png)

**Comment by wmbutler** [comment]
It might make sense to put it in the same box with My Open Orders. Since you can't see it where it is, anything is better than the current position.

**Comment by startailcoon** [comment]
If everyone agrees that it should be moved in to the My Open Orders, I'll do the change.

**Comment by abitmore** [comment]
Can we have responsive layout design on this screen?

**Comment by wmbutler** [comment]
@abitmore I'd like to wait on the responsive design until we get tradingview integrated.

**Comment by startailcoon** [comment]
# New Layout Combined My Open Orders + Settle Orders
As suggested:
- [x] Join Settle orders with My open orders
- [x] Last selected tab will stay selected on market change.
- [x] Settlement Price adjusted for `settle_offset %`
- [x] Include both assets in the pair
- [x] Fix for "x minutes ago" problem when a settlement had already been sent to market
- [x] View full date instead of "x minutes".

Still has a few minor issues to fix before I push this, but this is how it looks now and I do hope it can be accepted.

![image](https://user-images.githubusercontent.com/12114550/35296288-db520968-007b-11e8-8ff6-ea0742afd25a.png)

![image](https://user-images.githubusercontent.com/12114550/35296324-f92b9850-007b-11e8-8582-d095e09e5895.png)

### Requested hours:
Additional hours: 1.5
Total hours: 7.5

**Comment by wmbutler** [comment]
I like it.

**Comment by startailcoon** [comment]
I think I've got the calculation correct, but just making sure it is I throw out the question here.

Asset: BTS/bitCNY
Offset: 5%

If the feed price is 0.29554, the settlement price should be this +5%, meaning 0.3103.
Have i got this right?

**Note that this is the old headers**

![image](https://user-images.githubusercontent.com/12114550/35486326-ff3faed0-046c-11e8-9442-4d00248bb78d.png)

![image](https://user-images.githubusercontent.com/12114550/35486318-d48c4cb6-046c-11e8-938c-1ec6b2eec995.png)

**Comment by wmbutler** [comment]
@abitmore can we have your input here?

**Comment by abitmore** [comment]
> If the feed price is 0.29554, the settlement price should be this +5%, meaning 0.3103.

This is incorrect. Feed price is 0.29554 BTS per CNY, or say 3.3836 CNY per BTS. Settlement price +5% means 3.3836*105% = 3.5528 CNY per BTS, or say 0.29554/1.05 = 0.2815 BTS per CNY.

**Comment by startailcoon** [comment]
Easy correction
